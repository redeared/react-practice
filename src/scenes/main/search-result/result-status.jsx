import React from 'react';

export const ResultStatus = ({ items }) => {
    if (!items) {
        return null;
    }
    return (
        <span> {items && items.length} movie{items && items.length > 1 && 's'} found </span>
    );
};
