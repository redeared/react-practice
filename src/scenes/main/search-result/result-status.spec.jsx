import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import { ResultStatus } from './result-status';

describe('ResultStatus', () => {
    it('should render without crash', () => {
        const comp = create(<ResultStatus items={['empty']} />).toJSON();
        expect(comp).toMatchSnapshot();
    });

    it('should not rendered without items', () => {
        const comp = shallow(<ResultStatus />);

        expect(comp.getElement()).toBe(null);
    });
});
