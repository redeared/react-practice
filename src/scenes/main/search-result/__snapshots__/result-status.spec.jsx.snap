// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultStatus should render without crash 1`] = `
<span>
   
  1
   movie
   found 
</span>
`;
